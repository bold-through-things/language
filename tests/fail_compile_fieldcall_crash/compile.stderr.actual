[
  {
    "recoverable": false,
    "line": 22,
    "char": 0,
    "content": "67lang:call heartbeat",
    "error": "failed to assert: could not find a matching overload for heartbeat with arguments []",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": []
  },
  {
    "recoverable": false,
    "line": 15,
    "char": 0,
    "content": "67lang:call event",
    "error": "failed to assert: could not find a matching overload for event with arguments []",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "field": "event",
        "demands": [
          "HTMLScriptElement"
        ],
        "returns": "str",
        "convention": "FieldCall"
      },
      {
        "field": "event",
        "demands": [
          "HTMLScriptElement",
          "str"
        ],
        "returns": "None",
        "convention": "FieldCall"
      },
      {
        "field": "event",
        "demands": [
          "Window"
        ],
        "returns": "Event",
        "convention": "FieldCall"
      }
    ]
  },
  {
    "recoverable": false,
    "line": 22,
    "char": 0,
    "content": "67lang:call heartbeat",
    "error": "failed to assert: could not find a matching overload for heartbeat with arguments None",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": []
  },
  {
    "recoverable": false,
    "line": 15,
    "char": 0,
    "content": "67lang:call event",
    "error": "failed to assert: could not find a matching overload for event with arguments None",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "field": "event",
        "demands": [
          "HTMLScriptElement"
        ],
        "returns": "str",
        "convention": "FieldCall"
      },
      {
        "field": "event",
        "demands": [
          "HTMLScriptElement",
          "str"
        ],
        "returns": "None",
        "convention": "FieldCall"
      },
      {
        "field": "event",
        "demands": [
          "Window"
        ],
        "returns": "Event",
        "convention": "FieldCall"
      }
    ]
  }
]
