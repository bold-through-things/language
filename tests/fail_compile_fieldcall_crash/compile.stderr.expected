[
  {
    "recoverable": false,
    "line": 22,
    "char": 0,
    "content": "67lang:call heartbeat",
    "error": "failed to assert: could not find a matching overload for heartbeat with arguments []",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": []
  },
  {
    "recoverable": false,
    "line": 15,
    "char": 0,
    "content": "67lang:call event",
    "error": "failed to assert: could not find a matching overload for event with arguments []",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "field": "event",
        "demands": [
          "HTMLScriptElement"
        ],
        "returns": {
          "name": "str"
        },
        "convention": "FieldCall"
      },
      {
        "field": "event",
        "demands": [
          "HTMLScriptElement",
          {
            "name": "str"
          }
        ],
        "returns": "None",
        "convention": "FieldCall"
      },
      {
        "field": "event",
        "demands": [
          {
            "name": "Window",
            "type_params": [],
            "fields": []
          }
        ],
        "returns": {
          "name": "Event",
          "type_params": [],
          "fields": []
        },
        "convention": "FieldCall"
      }
    ]
  },
  {
    "recoverable": false,
    "line": 22,
    "char": 0,
    "content": "67lang:call heartbeat",
    "error": "failed to assert: could not find a matching overload for heartbeat with arguments None",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": []
  },
  {
    "recoverable": false,
    "line": 15,
    "char": 0,
    "content": "67lang:call event",
    "error": "failed to assert: could not find a matching overload for event with arguments None",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "field": "event",
        "demands": [
          "HTMLScriptElement"
        ],
        "returns": {
          "name": "str"
        },
        "convention": "FieldCall"
      },
      {
        "field": "event",
        "demands": [
          "HTMLScriptElement",
          {
            "name": "str"
          }
        ],
        "returns": "None",
        "convention": "FieldCall"
      },
      {
        "field": "event",
        "demands": [
          {
            "name": "Window",
            "type_params": [],
            "fields": []
          }
        ],
        "returns": {
          "name": "Event",
          "type_params": [],
          "fields": []
        },
        "convention": "FieldCall"
      }
    ]
  }
]
